iron-overlay-behavior.js

```diff
@@ -763,6 +763,18 @@ export const IronOverlayBehaviorImpl = {
 
 };
 
+const composedParent = node =>
+    node.assignedSlot || node.parentNode || node.host;
```

Comment:
I'm on the fence of this is scary and this is brilliant, but it seems to terminate in 100% of the cases I can think of. I wonder if you can achieve this with event listeners...

Reply:
Actually no, it won't work with listeners in the case of

```html
<a-a>
  #root1
    <my-dialog>
      #root2
        <ancestor @listener=${thisWontBeCalled}>
          <slot></slot>
        </ancestor>
      #/root2

      <outerAncestor @listener=${thisWillBeCalled}>
        <slot></slot>
      </outerAncestor>
    </my-dialog>
  #/root1

  <div id="target"></div>
</a-a>
```

Reply:
example:

https://stackblitz.com/edit/typescript-litelement-example-fr9vvp?file=Cat.ts